### ISSUE TITLE
Building a HB template with `$CLASS$` text fails

### ISSUE THREAD

**Issue by jonschlinkert** [assemble/grunt-assemble#19]
moved from https://github.com/assemble/assemble/issues/580#issuecomment-101763972

**Comment by knight-dj-et** [comment]
https://github.com/assemble/grunt-assemble/blob/master/tasks/assemble.js#L700

1 line change

change `layout.replace(assemble.engine.bodyRegex, body);` to `layout.replace(assemble.engine.bodyRegex, body.replace("$", "$$$$"));`

**Comment by doowb** [comment]
@jonschlinkert's solution with `.split().join()` is probably a better way to go. The other way is to use `function () { return body; }` instead of `body`, but I think Jon's benchmarked the split/join and it's usually faster, right?

@knight-dj-et if you'd like to do a PR, that'd be awesome!

**Comment by jonschlinkert** [comment]
> I think Jon's benchmarked the split/join and it's usually faster, right?

well, it's not about faster here. it's only useful if we want to replace multiple occurrences of `body`, which I 'm almost positive we were doing - at least at one point. I remember the pr that added that. If not, then we can't do split.

**Comment by jonschlinkert** [comment]
but yes, it's also considerably faster

**Comment by jonschlinkert** [comment]
wait, why would we do `layout.replace(assemble.engine.bodyRegex, body.replace("$", "$$$$"));`. that will arbitrarily replace the first `$` character anywhere in the body string with `$$$$`.

**Comment by knight-dj-et** [comment]
Oh. I suppose I did it too broadly.  It should replace any $ followed by a $, ', `, & or number with "$$" (which has to be passed as "$$$$" since this is using replace which will still treat the $ as special characters)

**Comment by jonschlinkert** [comment]
we would need to do something like this:

```js
var injectBody = function(layout, body) {
  body = body.split('$').join('\\$');
  var res = layout.replace(assemble.engine.bodyRegex, body);
  return res.split('\\$').join('$');
};
```

**Comment by jonschlinkert** [comment]
the other alternative is to just do split/join on the actual body tag. IMO that's a better way to go. since : 1) a user would expect any tag to yield the same result if it was used more than once in a string, 2) it would only do the split/join operation once, whereas the escaping approach ends up doing it three times (including `.replace()`)

**Comment by doowb** [comment]
All we need to do is `layout.replace(assemble.engine.bodyRegex, function () { return body; });`

**Comment by jonschlinkert** [comment]
^

**Comment by jonschlinkert** [comment]
@doowb you said that earlier and I missed the function part of the replacement...

**Comment by doowb** [comment]
fixed in `grunt-assemble@0.3.1`
